src/main/kotlin/com/romainpiel/catsanddogs/ScheduleRepository.kt

```diff
@@ -25,20 +29,42 @@ class ScheduleRepository {
                 .create()
 
         val retrofit = Retrofit.Builder()
-                .baseUrl("https://catsanddogs-kotlin-server.herokuapp.com")
+                .baseUrl("https://catsanddogs-swift-server.herokuapp.com/")
                 .addConverterFactory(GsonConverterFactory.create(gson))
                 .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                 .build()
 
         service = retrofit.create(ScheduleService::class.java)
     }
 
-    fun schedule(from: OffsetDateTime?): Single<String> {
+    fun schedule(from: OffsetDateTime?, locale: Locale, conference: Conference): Single<String> {
         val safeFrom = from ?: OffsetDateTime.MIN
-        return service.getConference()
+
+        when (conference) {
+                Conference.MCE4 -> return scheduleMCE4(safeFrom, locale)
+                Conference.KotlinConf -> return scheduleKotlinConf(safeFrom, locale)
+        }
+    }
+
+    private fun scheduleMCE4(from: OffsetDateTime, locale: Locale): Single<String> {
```

Comment:
As we have a lot of shared code between the two methods, we could also do:
```kotlin
private fun Observable<ApiConference>.toJson(from: OffsetDateTime, locale: Locale) =
        this.map { it.schedule }
              .flatMapIterable { it }
              .filter { !safeFrom.isAfter(it.datestamp) }
              .filter { !from.isAfter(it.datestamp) }
              .map {
                  val date = it.datestamp.format(ofPattern("EEEE, d MMMM y", locale)).capitalize()
                  val time = it.datestamp.format(ofPattern("HH:mm", locale))
                  Item(it.title, it.speaker.joinToString(), date, time)
              }
              .toList()
              .toJson(gson)
```

and use it like that:
```kotlin
private fun scheduleMCE4(from: OffsetDateTime, locale: Locale) = service.getMCE4Schedule().toJson()
```

(not entirely sure the code compiles, I just wrote it here :D)

Reply:
Yep